packages/reket-core/src/response.js

```diff
@@ -1,116 +1,144 @@
 
-  /**
-   * The response headers returned by the HTTP call.
-   * @name ReketResponse#headers
-   * @type {Object}
-   * @private
-   */
-  #headers;
-
-  /**
-   * The original request that has been send.
-   * @name ReketResponse#request
-   * @type {Object}
-   * @private
-   */
-  #request;
+/**
+ * Creates a response with data exposed and other response attributes exposed as readonly.
+ * This format will avoid having to access `response.data`` in order to get the data.
+ * Imagine we have a HTTP with response promise as follow:
+ * ```
+ * "data": {
+ *   "shi": "ming",
+ *   "foo": "bar",
+ *  "me": "zot"
+ * },
+ * "headers": {
+ *   ...
+ * }
+ * ```
+ * To access data in the promise you have to do something like:
+ * ```
+ * Reket.get('/my/url').then((response) => {
+ *   const foo = response.data.foo;
+ *   // or const { foo } = response.data;
+ *   // and then access the headers
+ *   const headers = response.headers;
+ * });
+ * ```
+ *
+ * With ReketResponse format it's possible to get directly foo from response like so:
+ * ```
+ * Reket.get('/my/url').then((response) => {
+ *   const foo = response.foo;
+ *   // or const { foo } = response;
+ *   // and then access the headers
+ *   const headers = response.getHeaders;
+ * });
+ * ```
+ *
+ * @param  {*}      options.data       The data returned by the HTTP call.
+ * @param  {Object} options.config     The original config of the request.
+ * @param  {Object} options.headers    The response headers returned by the HTTP call.
+ * @param  {Object} options.request    The original request that has been send.
+ * @param  {string} options.status     The response status (e.g. 200, 304, ...)
+ * @param  {string} options.statusText The status text of the response (e.g. "OK",
+ *                                     "Not Modified", ...)
+ *
+ * @return {ReketResponse}
+ */
 
-  /**
-   * The response status (e.g. 200, 304, ...)
-   * @name ReketResponse#status
-   * @type {string}
-   * @private
-   */
-  #status;
 
-  /**
-   * The status text of the response (e.g. "OK", "Not Modified", ...)
-   * @name ReketResponse#statusText
-   * @type {string}
-   * @private
-   */
-  #statusText;
 
-  constructor(clientResponse) {
-    const { data } = clientResponse;
+export const buildReketResponse = ({ data, config, headers, request, status, statusText }) => {
+  let props = {};
+  let responseData = data;
+  let response = {};
 
-    if (data instanceof Object) {
-      Object.assign(this, data);
-    } else {
-      this.data = data;
+  // detect type of response data
+  if (!Array.isArray(responseData)) {
+    // if not an array (Object, string, number, ...)
+    if (typeof responseData !== 'object') {
+      // and not an object (number, string, null, ...).
+      // Treat it as an object and wrap data into an object with value as key.
+      // If HTTP data response is `"response"` the reket response object will look like:
+      //
+      // ```
+      // {
+      //   value: 'response'
```

Comment:
don't know if it's a good idea or not.

Reply:
So you're checking the different types to avoid errors due to the Object.keys? 

Maybe instead of checking everything with if, you could just do a try catch ? 

```
try {
  // transform your object normally 
} catch (e) {
  // Can't be transformed because of the wrong type
  // responseData = { value : ... } 
}
```

You could also check the error if u want to be more precise about it

Reply:
Mmmmh, for example if I have `response = { data: 2 }`, then if I use `Object.keys(response.data)` I will have an empty array, so I don't think that `try/catch` will work here.

I wonder here if it's a good idea to transform response data like so: if it's not a "compatible" type, then transform it to an object with `value` as key. This is already what we are doing sometimes in ovh-api-services for example : 

https://github.com/ovh-ux/ovh-api-services/blob/55dac6f536d0d70a3360e4e8655ea41e881cfb82/src/api/telephony/telephony.v6.service.js#L116-L123